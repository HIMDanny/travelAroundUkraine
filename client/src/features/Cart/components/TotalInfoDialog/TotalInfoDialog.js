import React, { useState, memo } from 'react';
import PropTypes from 'prop-types';
import { useNavigate } from 'react-router-dom';

// MUI
import { Button, Stack, Box as MuiBox, Typography } from '@mui/material';
import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
import { BoxWrapper, Section, Title, Select, MenuItem } from '../../../../components';

// Utils & Data
import calcToursQuantityAndPrice from '../../utils/calcToursQuantityAndPrice';
import currencyConverter from '../../../Tour/utils/currencyConverter';
import currencySymbols from '../../../data/currencySymbols';

const TotalInfoDialog = ({ cart }) => {
  const navigate = useNavigate();
  const [currency, setCurrency] = useState('eur');
  let toursPriceAndQuantity;

  if (cart.length) {
    toursPriceAndQuantity = calcToursQuantityAndPrice(cart);
  }

  const onCurrencyChange = (e) => {
    setCurrency(e.target.value);
  };

  return (
    <BoxWrapper>
      <Section>
        <Title>Total</Title>
        <Stack direction="row" justifyContent="space-between">
          <Typography>Tours amount:</Typography>
          <Typography>{toursPriceAndQuantity?.toursQuantity}</Typography>
        </Stack>
      </Section>

      <Section>
        <Title>Currency</Title>
        <Select fullWidth value={currency} onChange={onCurrencyChange} IconComponent={ExpandMoreIcon}>
          {Object.keys(currencySymbols).map((key) => (
            <MenuItem key={key} value={key}>
              {key.toUpperCase()}
            </MenuItem>
          ))}
        </Select>
      </Section>

      <Section>
        <Stack direction="row" justifyContent="space-between">
          <Typography>To be paid:</Typography>
          <Typography fontSize={18}>
            {currencySymbols[currency]}
            {currencyConverter(toursPriceAndQuantity?.toursPrice, currency)}
          </Typography>
        </Stack>
      </Section>

      <MuiBox>
        <Stack direction="row" justifyContent="end" alignItems="center">
          <Button sx={{ paddingInline: '30px' }} disableElevation onClick={() => navigate('/checkout')}>
            Checkout
          </Button>
        </Stack>
      </MuiBox>
    </BoxWrapper>
  );
};

TotalInfoDialog.propTypes = {
  cart: PropTypes.array,
};

TotalInfoDialog.defaultProps = {
  cart: [],
};

export default memo(TotalInfoDialog);
