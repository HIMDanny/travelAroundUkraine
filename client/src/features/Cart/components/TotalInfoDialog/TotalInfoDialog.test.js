/* eslint-disable react/prop-types */
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import TotalInfoDialog from './TotalInfoDialog';

const RouterWrapper = ({ children }) => <MemoryRouter>{children}</MemoryRouter>;

jest.mock('@mui/material/Button', () => ({ children }) => <button type="button">{children}</button>);
jest.mock('@mui/material/Stack', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/Box', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/Typography', () => ({ children }) => <p>{children}</p>);

describe('TotalInfoDialog snapshot test', () => {
  test('should TotalInfoDialog match snapshot', () => {
    const { asFragment } = render(<TotalInfoDialog />, { wrapper: RouterWrapper });

    expect(asFragment()).toMatchSnapshot();
  });
});
