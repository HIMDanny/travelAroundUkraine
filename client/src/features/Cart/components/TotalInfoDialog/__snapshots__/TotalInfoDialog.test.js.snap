// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TotalInfoDialog snapshot test should TotalInfoDialog match snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <p>
        Total
      </p>
      <div>
        <p>
          Tours amount:
        </p>
        <p />
      </div>
    </div>
    <div>
      <p>
        Currency
      </p>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth css-rtvg6-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
          role="button"
          tabindex="0"
        >
          EUR
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          tabindex="-1"
          value="eur"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ExpandMoreIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ihdtdm"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div>
      <div>
        <p>
          To be paid:
        </p>
        <p>
          €
        </p>
      </div>
    </div>
    <div>
      <div>
        <button
          type="button"
        >
          Checkout
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
